lightning/src/ln/peer_handler.rs

```diff
@@ -389,7 +389,13 @@ const BUFFER_DRAIN_MSGS_PER_TICK: usize = 32;
 
 struct Peer {
 	channel_encryptor: PeerChannelEncryptor,
+	// When setting this field to a `Some(..)` value, use `Peer::set_their_node_id` to also set
+	// the following field `their_node_id_serialized`.
 	their_node_id: Option<PublicKey>,
+	// This field is a cached `NodeId` of `their_node_id` to avoid serializing peers' keys every
+	// time we forward gossip messages in `PeerManager`. It should be set anytime `their_node_id`
+	// gets set.
+	their_node_id_serialized: Option<NodeId>,
```

Comment:
I'm a bit worried these could go out of sync, but doc comments which would at least appear in most author IDEs could help with that if there's no better way (which I can't think of right now). Unfortunately, single fields cannot be made externally immutable in Rust. I'm sure others might have some experience here. :)

```suggestion
	/// NOTE: Do not modify this field directly. Use `Peer::set_their_node_id` instead.
	their_node_id: Option<PublicKey>,
	/// This field is a cached `NodeId` of `their_node_id` to avoid serializing peers' keys every
	/// time we forward gossip messages in `PeerManager`.
	///
	/// NOTE: Do not modify this field directly. Use `Peer::set_their_node_id` instead.
	their_node_id_serialized: Option<NodeId>,
```

Reply:
Oh oops I meant to make them doc comments, thanks. And yea those were exactly my thoughts, at the moment I can't think of a way to cache without risking them going out of sync, hoping this would have the least friction for keeping them in sync.

Reply:
Aha! Simple :)

Reply:
Oh nice! Will update